File: employee/src/main/java/io/nology/employee/Employee/UpdateEmployeeDTO.java
package io.nology.employee.Employee;

import java.util.Date;

import org.hibernate.validator.constraints.Length;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.format.annotation.DateTimeFormat.ISO;

import jakarta.annotation.Nullable;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;

public class UpdateEmployeeDTO {

    @Pattern(regexp = ".*\\S.*", message = "First Name cannot be empty")
    @Length(min = 2)
    private String firstName;
    @Nullable
    private String middleName;
    @Pattern(regexp = ".*\\S.*", message = "Last Name cannot be empty")
    @Length(min = 2)
    private String lastName;

    @Pattern(regexp = "male\\b|female\\b|nonbinary\\b", message = "input must be male, female or nonbinary")
    private String gender;
    @DateTimeFormat(iso = ISO.DATE, fallbackPatterns = { "M/d/yy", "dd.MM.yyyy" })
    private Date birthDate;
  
    @Email
    @Pattern(regexp = ".*\\S.*", message = "Email cannot be empty")
    private String email;
    // @NotBlank
    @Pattern(regexp = "\\d+", message = "only use numbers")
    @Pattern(regexp = ".*\\S.*", message = "Mobile cannot be empty")
    private String mobile;

    @Pattern(regexp = ".*\\S.*", message = "Street cannot be empty")
    private String street;

    @Pattern(regexp = ".*\\S.*", message = "Suburb cannot be empty")
    private String suburb;

    @Pattern(regexp = ".*\\S.*", message = "State cannot be empty")
    private String state;

    @Pattern(regexp = "\\d+", message = "only use numbers")
    @Pattern(regexp = ".*\\S.*", message = "Postcode cannot be empty")
    private String postCode;
 
    private Boolean isPermanent;

    private Boolean isFullTime;
    @DateTimeFormat(iso = ISO.DATE, fallbackPatterns = { "M/d/yy", "dd.MM.yyyy" })
    private Date startDate;
    @DateTimeFormat(iso = ISO.DATE, fallbackPatterns = { "M/d/yy", "dd.MM.yyyy" })
    private Date finishDate;
    @NotNull
    private double weeklyHours;
    public String getFirstName() {
        return firstName;
    }
    public String getMiddleName() {
        return middleName;
    }
    public String getLastName() {
        return lastName;
    }
    public String getGender() {
        return gender;
    }
    public Date getBirthDate() {
        return birthDate;
    }
    public String getEmail() {
        return email;
    }
    public String getMobile() {
        return mobile;
    }
    public String getStreet() {
        return street;
    }
    public String getSuburb() {
        return suburb;
    }
    public String getState() {
        return state;
    }
    public String getPostCode() {
        return postCode;
    }
    public Boolean getIsPermanent() {
        return isPermanent;
    }
    public Boolean getIsFullTime() {
        return isFullTime;
    }
    public Date getStartDate() {
        return startDate;
    }
    public Date getFinishDate() {
        return finishDate;
    }
    public double getWeeklyHours() {
        return weeklyHours;
    }

    

}

File: CsharpEmployee/Employee/DTOs/UpdateEmployeeDto.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;

namespace CsharpEmployee.Employee.DTOs
{
    public class UpdateEmployeeDto
    {
        [Required(ErrorMessage = "First Name is required")]
        [StringLength(50, ErrorMessage = "First Name cannot be longer than 50 characters")]
        public required string FirstName { get; set; }

        [MaybeNull]
        [StringLength(50, ErrorMessage = "Middle Name cannot be longer than 50 characters")]
        public string? MiddleName { get; set; }
        [Required(ErrorMessage = "Last Name is required")]
        [StringLength(50, ErrorMessage = "Last Name cannot be longer than 50 characters")]
        public required string LastName { get; set; }

        [Required(ErrorMessage = "Gender is required")]
        [RegularExpression(@"^(male|female|nonbinary)$", ErrorMessage = "Gender must be male, female, or nonbinary")]
        public required string Gender { get; set; }

        [MaybeNull]
        public DateOnly? DateOfBirth { get; set; }

        [Required(ErrorMessage = "Email is required")]
        [EmailAddress(ErrorMessage = "Invalid email address")]
        public required string Email { get; set; }

        [Required(ErrorMessage = "Mobile number is required")]
        [RegularExpression(@"^\d+$", ErrorMessage = "Mobile number must contain only digits")]
        public required string Mobile { get; set; }

                [Required(ErrorMessage = "Street is required")]
        [StringLength(100, ErrorMessage = "Street cannot be longer than 100 characters")]
        public required string Street {get; set;}

        [Required(ErrorMessage = "Suburb is required")]
        [StringLength(100, ErrorMessage = "Suburb cannot be longer than 100 characters")]
        public required string Suburb {get; set;}

        [Required(ErrorMessage = "State is required")]
        [StringLength(50, ErrorMessage = "State cannot be longer than 50 characters")]
        public required string State {get; set;}

        [Required(ErrorMessage = "Postcode is required")]
        [StringLength(4, ErrorMessage = "Postcode cannot be longer than 4 characters")]
        [RegularExpression(@"^\d+$", ErrorMessage = "Postcode must contain only digits")]
        public required string Postcode {get; set;}

    }
}